src/python/pants/engine/http.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Dict, Optional, Tuple
+
+import requests
+
+from pants.engine.rules import side_effecting
+
+
+@dataclass(frozen=True)
+class HttpGetResponse:
+  url: str
+  status_code: Optional[int]
+  output_bytes: Optional[bytes]
+  headers: Tuple[Tuple[str, str], ...]
+
+
+@side_effecting
+class HttpRequester:
```

Comment:
Should this be a dataclass? It's possible for dataclasses with no fields. Why go through the effort?

1) Nice defaults for things like `__repr__`.
2) `frozen=True` means that this will do what we want for `__hash__`.
3) Were we to ever add any fields, it's seamless to do.
4) Consistency with the rest of the codebase.